File: csv/src/main/scala/akka/stream/alpakka/csv/scaladsl/ByteOrderMark.scala
/*
 * Copyright (C) since 2016 Lightbend Inc. <https://www.lightbend.com>
 */

package akka.stream.alpakka.csv.scaladsl

import akka.util.ByteString

/**
 * Byte Order Marks may be used to indicate the used character encoding
 * in text files.
 *
 * @see https://www.unicode.org/faq/utf_bom.html#bom1
 */
object ByteOrderMark {

  private[this] final val ZeroZero = ByteString.apply(0x00.toByte, 0x00.toByte)

  /** Byte Order Mark for UTF-16 big-endian */
  final val UTF_16_BE = ByteString.apply(0xfe.toByte, 0xff.toByte)

  /** Byte Order Mark for UTF-16 little-endian */
  final val UTF_16_LE = ByteString.apply(0xff.toByte, 0xfe.toByte)

  /** Byte Order Mark for UTF-32 big-endian */
  final val UTF_32_BE = ZeroZero ++ UTF_16_BE

  /** Byte Order Mark for UTF-32 little-endian */
  final val UTF_32_LE = UTF_16_LE ++ ZeroZero

  /** Byte Order Mark for UTF-8 */
  final val UTF_8 = ByteString.apply(0xef.toByte, 0xbb.toByte, 0xbf.toByte)
}

File: csv/src/main/java/akka/stream/alpakka/csv/javadsl/ByteOrderMark.java
/*
 * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
 */

package akka.stream.alpakka.csv.javadsl;

import akka.util.ByteString;

/**
 * Byte Order Marks may be used to indicate the used character encoding in text files.
 *
 * @see <a href="https://www.unicode.org/faq/utf_bom.html#bom1">Unicode FAQ Byte Order Mark</a>
 */
public class ByteOrderMark {
  /** Byte Order Mark for UTF-16 big-endian */
  public static final ByteString UTF_16_BE =
      akka.stream.alpakka.csv.scaladsl.ByteOrderMark.UTF_16_BE();

  /** Byte Order Mark for UTF-16 little-endian */
  public static final ByteString UTF_16_LE =
      akka.stream.alpakka.csv.scaladsl.ByteOrderMark.UTF_16_LE();

  /** Byte Order Mark for UTF-32 big-endian */
  public static final ByteString UTF_32_BE =
      akka.stream.alpakka.csv.scaladsl.ByteOrderMark.UTF_32_BE();

  /** Byte Order Mark for UTF-32 little-endian */
  public static final ByteString UTF_32_LE =
      akka.stream.alpakka.csv.scaladsl.ByteOrderMark.UTF_32_LE();

  /** Byte Order Mark for UTF-8 */
  public static final ByteString UTF_8 = akka.stream.alpakka.csv.scaladsl.ByteOrderMark.UTF_8();
}
